lib/tasks/fix_cached_counters.rake

```diff
@@ -1,24 +1,11 @@
 desc 'Fix cached counters by reseting them to the correct count, in case they got corrupted somehow'
 task :fix_cached_counters => :environment do
   puts "Correcting cached counters: (this may take a few minutes)"
-  [ Architecture, Environment, Operatingsystem, Realm].each do |cl|
-    cl.unscoped.all.each{|el| cl.reset_counters(el.id, :hosts, :hostgroups)}
-    puts "#{cl} corrected"
-  end
-  Domain.all.each do |el|
-    Domain.reset_counters(el.id, :hostgroups)
-    el.update_attribute('total_hosts', Nic::Base.primary.where(:domain_id => el.id).count)
-  end
-  puts "Domains corrected"
   Puppetclass.all.each{|el| Puppetclass.reset_counters(el.id, :lookup_keys)}
   puts "Puppetclass corrected"
-  Model.all.each{|el| Model.reset_counters(el.id, :hosts)}
-  puts "Model corrected"
   ConfigGroup.all.each{|el| ConfigGroup.reset_counters(el.id, :config_group_classes)}
   puts "ConfigGroup corrected"
   LookupKey.all.each{|el| LookupKey.reset_counters(el.id, :lookup_values)}
   puts "LookupKey corrected"
-  Hostgroup.all.each{|el| Hostgroup.reset_counters(el.id, :hosts)}
-  puts "Hostgroup corrected"
 end
```

Comment:
> yes, the puppet class related counters don't face the same issue with deadlocks as the host ones, but we can consider getting rid of them as well. Not sure it has to be in the same pr.

I think for consistency, we should do it in one PR. Just rename issue/PR
and go for it. The theme is simply dropping all cache counters from
here.

I will also do testing in beaker against 100+ real hosts for this patch.

-- 
Later,
 Lukas #lzap Zapletal